utilities/blob_db/blob_db_listener.h

```diff
@@ -37,10 +37,30 @@ class BlobDBListener : public EventListener {
     blob_db_impl_->UpdateLiveSSTSize();
   }
 
- private:
+ protected:
   BlobDBImpl* blob_db_impl_;
 };
 
+class BlobDBListenerGC : public BlobDBListener {
```

Comment:
I'm picky but I still prefer composition to inheritance, as what Google C++ Style suggests: https://google.github.io/styleguide/cppguide.html#Inheritance

In this example, I don't see a reason we can't composite.

Reply:
I agree that in many cases composition is a better idea but in this specific case, I think it would be very counterintuitive for one subtype of `EventListener` to have another type of `EventListener` as a member. Also, if we were to do that, `BlobDBListenerGC` would have to implement `OnFlushBegin` just so it can dispatch to the other listener.